src/twisted/python/_setup.py

```diff
@@ -96,6 +96,7 @@
         'pyasn1',
         'cryptography >= 0.9.1',
         'appdirs >= 1.4.0',
+        'pynacl >= 1.0.1',
```

Comment:
Good question. I've never been a huge fan of optional deps, but a project like twisted probably does need to care about things like that. It doesn't seem unreasonable to state that a base install of twisted conch must support RSA and NIST ecdsa (via a non-optional cryptography dep), but that users who want access to ed25519 should install pynacl. ed25519 is likely to appear in cryptography soon-ish as well, albeit only accessible to users linking against the latest OpenSSL.